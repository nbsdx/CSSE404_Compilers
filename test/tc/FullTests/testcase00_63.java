/*
 * Testcase designed to test expression error checking.
 * This should not pass, and the compiler should give errors to the effect of
 * "type mismatch in expression".
 */

class Test6 {
	public static void main(String[] args) {
		{
			// errors here
			int x = 3 + true;
			int y = x + 3 * (4 / false) + x;
			int z = true - (true && 18) || y;
		}
		// x no longer exists here
		int z = x; 
	}
}

// vim:filetype=java:textwidth=80
