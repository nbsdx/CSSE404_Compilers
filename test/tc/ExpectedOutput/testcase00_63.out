Right argument of type boolean does not match expected type int for operator PLUS
Right argument of type boolean does not match expected type int for operator DIVIDE
Right argument of type int does not match expected type boolean for operator AND
Left argument of type boolean does not match expected type int for operator MINUS
Right argument of type boolean does not match expected type int for operator MINUS
Left argument of type int does not match expected type boolean for operator OR
Right argument of type int does not match expected type boolean for operator OR
Cannot assign type boolean to variable z of type int
Variable x is not declared.
Cannot assign type null to variable z of type int
